src/promptflow/tests/executor/e2etests/test_traces.py

```diff
@@ -527,6 +596,67 @@ def test_flow_with_nested_tool(self):
+        flow_folder = get_flow_folder(flow_file)
+        if (span_folder := flow_folder / ".span").exists():
+            for file in span_folder.glob("*.jsonl"):
+                file.unlink()
+        else:
+            span_folder.mkdir()
+
+        with override_process_pool_targets(process_manager=mock_process_manager, process_wrapper=mock_process_wrapper):
+            mem_run_storage = MemoryRunStorage()
+            batch_result = submit_batch_run(
+                flow_folder=flow_file,
+                inputs_mapping={"user_id": "${data.user_id}"},
+                input_file_name="inputs.jsonl",
+                connections=dev_connections,
+                storage=mem_run_storage,
+            )
+            assert isinstance(batch_result, BatchResult)
+
+            batch_run_id = list(mem_run_storage._flow_runs.values())[0].root_run_id
+            assert (flow_folder / ".span").exists()
+            trace_ids = []
+            for file in span_folder.glob("*.jsonl"):
+                spans = []
+                with open(file, "r") as f:
+                    json_chunks = f.read().strip().split("\n\n")
+                    for chunk in json_chunks:
+                        spans.append(json.loads(chunk))
+                trace_ids.append(spans[0]["context"]["trace_id"])
+                assert len(spans) == 5, f"Got {len(spans)} spans."
+                root_spans = [span for span in spans if span["parent_id"] is None]
+                assert len(root_spans) == 1
+                root_span = root_spans[0]
+                for span in spans:
+                    span["status"]["status_code"] = "OK"
+                    span["attributes"]["batch_run_id"] = batch_run_id
+                    span["attributes"]["framework"] = "promptflow"
+                    if span["parent_id"] is None:
+                        expected_span_type = "Flow"
+                    elif span["attributes"].get("function", "") in LLM_FUNCTION_NAMES:
+                        expected_span_type = "LLM"
+                    elif span["attributes"].get("function", "") in EMBEDDING_FUNCTION_NAMES:
+                        expected_span_type = "Embedding"
+                    else:
+                        expected_span_type = "Function"
+                    msg = f"span_type: {span['attributes']['span_type']}, expected: {expected_span_type}"
+                    assert span["attributes"]["span_type"] == expected_span_type, msg
+                    if span != root_span:  # Non-root spans should have a parent
+                        assert span["attributes"]["function"]
+                    inputs = json.loads(span["attributes"]["inputs"])
+                    output = json.loads(span["attributes"]["output"])
+                    assert isinstance(inputs, dict)
+                    assert output is not None
+
+            for run_info in mem_run_storage._flow_runs.values():
+                assert f"0x{int(run_info.otel_trace_id, 16):032x}" in trace_ids
```

Comment:
We may directly use `0x + run_info.otel_trace_id`

Reply:
Will update in next PR.